state/controller.go

```diff
@@ -142,3 +149,26 @@ func checkUpdateControllerConfig(name string) error {
 	}
 	return nil
 }
+
+// checkSpaceIsValid checks if each controller machine has addresses in the
+// the input space. If not, an error is returned.
+func (st *State) checkSpaceIsValid(configSpace string) error {
+	info, err := st.ControllerInfo()
+	if err != nil {
+		return errors.Annotate(err, "cannot get controller info")
+	}
+
+	spaceName := network.SpaceName(configSpace)
+	for _, id := range info.MachineIds {
+		m, err := st.Machine(id)
+		if err != nil {
+			return errors.Annotate(err, "cannot get machine")
+		}
+
+		if _, ok := network.SelectAddressesBySpaceNames(m.MachineAddresses(), spaceName); !ok {
+			return errors.Errorf("machine %q has no addresses in space", id)
```

Comment:
would it be nicer for operators to get an aggregated list, in case more than one machine is invalid? (rather than only the first)

Not sure it really matters, just wondering.

Reply:
Easy enough to do, so done.